newIDE/app/src/Profile/AuthenticatedUserProvider.js

```diff
@@ -75,12 +78,44 @@ export default class AuthenticatedUserProvider extends React.Component<
         onEdit: () => this.openEditProfileDialog(true),
         onCreateAccount: () => this.openCreateAccountDialog(true),
         onRefreshUserProfile: this._fetchUserProfile,
+        onRefreshFirebaseProfile: this._reloadFirebaseProfile,
+        onSendEmailVerification: this._doSendEmailVerification,
         getAuthorizationHeader: () =>
           this.props.authentication.getAuthorizationHeader(),
       },
     });
   }
 
+  _reloadFirebaseProfile = () => {
+    const { authentication } = this.props;
+
+    authentication.getFirebaseUser((err, firebaseUser: ?FirebaseUser) => {
+      if (err && err.unauthenticated) {
+        return this.setState({
+          authenticatedUser: {
+            ...this.state.authenticatedUser,
+            authenticated: false,
+            profile: null,
+            usages: null,
+            limits: null,
+            subscription: null,
+          },
+        });
+      } else if (err || !firebaseUser) {
+        console.log('Unable to fetch user profile', err);
+        return;
+      }
+
+      this.setState({
```

Comment:
Use `({authenticatedUser}) => ({authenticatedUser: {...authenticatedUser, ...}})` form which is a bit verbose but will ensure you don't erase the state with an old one.